### ISSUE TITLE
Question about refresh-token

### ISSUE THREAD

**Issue by tomc3** [layer7be/vue-starter-laravel-api#2]
I have two questions.

* Why we are generating new refresh-token after every request. Is there any security reasons or something else ?

* How to generate or get new token if for some reasons client app is not able to catch the new generated token. In that case user have to re-login to get new token again. ?

**Comment by kcalliauw** [comment]
Hi,

The single token flow is purely for security. The window of opportunity for an eavesdropper is so small that it would be very unlikely that a token could be used to successfully authenticate against the app. If you're not using TLS, you want the lifetime of a certain token to be as short as possible.

If the token is not stored then the app won't be able to make API calls and get a 401 Unauthorized reply back from, hence redirecting the user to the login page.

Hope that helps,
K

**Comment by tomc3** [comment]
Many thanks :)

**Comment by bkintanar** [comment]
Hi there. I seem to have a bug on my instance with your code. I've cloned both this and the Vue Companion. Whenever I refresh the page twice. I get logged out. But it doesn't happen in your demo site. Is there a code that wasn't pushed from the demo? I tried changing the

```    'blacklist_enabled' => env('JWT_BLACKLIST_ENABLED', true),```

to

```    'blacklist_enabled' => env('JWT_BLACKLIST_ENABLED', false),```

and it solves the problem. But I don't think that blacklist is meant to be disabled.
